src/python/pants/engine/process.py

```diff
@@ -50,7 +48,7 @@ def __init__(
         env: Optional[Dict[str, str]] = None,
         output_files: Optional[Tuple[str, ...]] = None,
         output_directories: Optional[Tuple[str, ...]] = None,
-        timeout_seconds: Union[int, float] = _default_timeout_seconds,
+        timeout_seconds: Optional[Union[int, float]] = None,
```

Comment:
Good change. Also, why can this be a float?

Reply:
For fractional seconds... it's fairly common in Python to use `float` for time. If anything, we could drop `int` here.